src/meteors/shap/explanation.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+from typing_extensions import Annotated
+from pydantic.functional_validators import PlainValidator
+from pydantic import ConfigDict, Field, ValidationInfo, BaseModel
+
+import numpy as np
+import pandas as pd
+import torch
+
+from loguru import logger
+
+from meteors.exceptions import ShapeMismatchError
+
+import shap
+
+AVAILABLE_SHAP_EXPLAINERS = ["exact", "tree", "kernel", "linear"]
+
+
+###################################################################
+########################### VALIDATIONS ###########################
+###################################################################
+
+
+def ensure_explainer_type(explainer_type: str | None) -> str | None:
+    explainer_type = explainer_type.lower() if explainer_type is not None else None
+    if explainer_type not in AVAILABLE_SHAP_EXPLAINERS and explainer_type is not None:
+        logger.warning(f"Invalid explainer type: {explainer_type}. ")
```

Comment:
I am not sure if it is an error, if yes we need to throw an error if not why warning?

Reply:
After analysing I think that we don't need to have this as it ensured in `shap.py`. Only question is if we need to have the None?

Reply:
fixed in ac31a16ee92f0ce12b39d88da40cd1329e6c9220

Removed the checks for the explainer type in the explanation.